uxarray/grid/grid.py

```diff
@@ -7,87 +7,108 @@
+    ``ux.UxDataset`` and accessed through the ``.uxgrid`` attribute.
+
+    For constructing a grid from non-UGRID datasets or other types of supported data, see our ``ux.open_grid`` method or
+    specific class methods (``Grid.from_dataset``, ``Grid.from_face_verticies``, etc.)
 
-    Can be used standalone to explore an unstructured grid topology, or
-    can be seen as the property of ``uxarray.UxDataset`` and ``uxarray.DataArray``
-    to make them unstructured grid-aware data sets and arrays.
 
     Parameters
     ----------
-    input_obj : xarray.Dataset, ndarray, list, tuple, required
-        Input ``xarray.Dataset`` or vertex coordinates that form faces.
-
-    Other Parameters
-    ----------------
-    gridspec: bool, optional
-        Specifies gridspec
-    islatlon : bool, optional
-        Specify if the grid is lat/lon based
-    isconcave: bool, optional
-        Specify if this grid has concave elements (internal checks for this are possible)
-    use_dual: bool, optional
-        Specify whether to use the primal (use_dual=False) or dual (use_dual=True) mesh if the file type is mpas
-
-    Raises
-    ------
-        RuntimeError
-            If specified file not found or recognized
+    grid_ds : xr.Dataset
+        ``xarray.Dataset`` encoded in the UGRID conventions
+
+    source_grid_spec : str, default="UGRID"
+        Original unstructured grid format (i.e. UGRID, MPAS, etc.)
+
+    source_dims_dict : dict, default={}
+        Mapping of dimensions from the source dataset to their UGRID equivalent (i.e. {nCell : nMesh2_face})
 
     Examples
     ----------
 
     >>> import uxarray as ux
+    >>> grid_path = "/path/to/grid.nc"
+    >>> data_path = "/path/to/data.nc"
 
     1. Open a grid file with `uxarray.open_grid()`:
 
-    >>> uxgrid = ux.open_grid("filename.g")
+    >>> uxgrid = ux.open_grid(grid_path)
 
     2. Open an unstructured grid dataset file with
-    `uxarray.open_dataset()`, then access `Grid` info:
+    `uxarray.open_dataset()`, then access the ``Grid``.:
 
-    >>> uxds = ux.open_dataset("filename.g")
+    >>> uxds = ux.open_dataset(grid_path, data_path)
+    >>> uxds.uxgrid
     """
 
-    def __init__(self, input_obj, **kwargs):
-        # initialize internal variable names
-        self.__init_grid_var_names__()
+    def __init__(self,
+                 grid_ds: xr.Dataset,
+                 source_grid_spec: Optional[str] = None,
+                 source_dims_dict: Optional[dict] = {}):
+
+        # grid spec not provided, check if grid_ds is a minimum representable UGRID dataset
+        if source_grid_spec is None:
+            warn(
+                "Attempting to construct a Grid without passing in source_grid_spec. Use of Grid constructor"
+                "is only advised if grid_ds is following the internal unstructured grid definition, including"
+                "variable and dimension names. Using ux.open_grid() or ux.from_dataset() is suggested.",
+                Warning)
```

Comment:
@erogluorhan 

I've added this warning, is there anything you'd suggest to add to better reflect the changes to the constructor?

Reply:
This looks great!